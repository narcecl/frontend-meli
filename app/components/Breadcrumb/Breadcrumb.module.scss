.breadcrumb{
    padding: 16px 0;
    margin: 0;
    display: flex;
    align-items: center;

    li{
        font-size: 0.875em;
        color: #999;

        svg{
            margin: 0 4px;
        }
    }
}