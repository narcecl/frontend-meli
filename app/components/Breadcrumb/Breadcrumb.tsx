import styles from './Breadcrumb.module.scss';
import ChevronRight from '../ChevronRight/ChevronRight';

export interface BreadcrumbProps {
    items: string[];
}

export const Breadcrumb = (props: BreadcrumbProps) => {
    const { items } = props;

    return (
        items?.length ?
        <ol className={styles.breadcrumb}>
            { items.map((item, i) => 
                <li key={i}>
                    { item }
                    { i < (items.length-1) ? <ChevronRight /> : null }
                </li>
            )}
        </ol>
        : null
    );
}

export default Breadcrumb;