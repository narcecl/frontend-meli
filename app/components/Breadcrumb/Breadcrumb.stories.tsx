import type { Meta, StoryObj } from '@storybook/react';
import { Breadcrumb } from './Breadcrumb';

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
const meta: Meta<typeof Breadcrumb> = {
    title: 'Core/Breadcrumb',
    component: Breadcrumb,
    tags: ['autodocs'],
    argTypes: {
        items: {
            description: 'Array of navigation items.'
        }
    }
};

export default meta;
type Story = StoryObj<typeof Breadcrumb>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default: Story = {
    args: {
        items: ['Elemento 1', 'Elemento 2', 'Elemento 3']
    },
};