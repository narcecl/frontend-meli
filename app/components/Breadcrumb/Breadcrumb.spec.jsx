import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Breadcrumb from './Breadcrumb';

describe('Breadcrumb', () => {
	it('Default render', () => {
		// Define the items for the breadcrumb
		const items = ['Category 1', 'Category 2', 'Category 3', 'Category 4'];

		// Render the component
		const { container, asFragment } = render(
			<Breadcrumb items={items} />
		);

		// Check first render
		const firstRender = asFragment();
        expect(firstRender).toMatchSnapshot();

		// Check if the breadcrumb items are rendered correctly
		const breadcrumbItems = screen.getAllByRole('listitem');
		expect(breadcrumbItems).toHaveLength(items.length);

		for( let i = 0; i < items.length; i++ ){
			// Check if breadcrumb item match the element from array
			const breadcrumbItem = breadcrumbItems[i];
			expect(breadcrumbItem).toHaveTextContent(items[i]);
		}

		// Check the presence of ChevronRight for all items except the last one
		const chevronRights = container.querySelectorAll('svg');
		expect(chevronRights).toHaveLength(items.length - 1);
	});

	it('Renders with no items', () => {
		// Render the component with no items
		const { container, asFragment } = render(<Breadcrumb items={[]} />);

		// Check first render
		const firstRender = asFragment();
        expect(firstRender).toMatchSnapshot();

		// Assert no breadcrumb items are rendered
		const breadcrumbItems = screen.queryAllByRole('listitem');
		expect(breadcrumbItems).toHaveLength(0);

		// Assert no ChevronRight is rendered
		const chevronRights = container.querySelectorAll('svg');
		expect(chevronRights).toHaveLength(0);
	});
});
