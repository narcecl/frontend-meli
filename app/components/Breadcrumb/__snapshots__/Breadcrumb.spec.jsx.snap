// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Default render 1`] = `
<DocumentFragment>
  <ol
    class="breadcrumb"
  >
    <li>
      Category 1
      <svg
        aria-hidden="true"
        class="chevronRight"
        fill="currentColor"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
          fill-rule="evenodd"
        />
      </svg>
    </li>
    <li>
      Category 2
      <svg
        aria-hidden="true"
        class="chevronRight"
        fill="currentColor"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
          fill-rule="evenodd"
        />
      </svg>
    </li>
    <li>
      Category 3
      <svg
        aria-hidden="true"
        class="chevronRight"
        fill="currentColor"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
          fill-rule="evenodd"
        />
      </svg>
    </li>
    <li>
      Category 4
    </li>
  </ol>
</DocumentFragment>
`;

exports[`Breadcrumb Renders with no items 1`] = `
<DocumentFragment>
  <ol
    class="breadcrumb"
  />
</DocumentFragment>
`;
